dota2/client.py

```diff
@@ -150,17 +153,21 @@ def send(self, emsg, data={}, proto=None):
         :type data: :class:`dict`
         :param proto: (optional) manually specify protobuf, other it's detected based on ``emsg``
         """
+        data = {} if data is None else data
         self._send(emsg, data, proto)
 
-    def _send(self, emsg, data={}, proto=None, jobid=None):
+    def _send(self, emsg, data=None, proto=None, jobid=None):
```

Comment:
This is not needed. The warning is correct that you should not assign a reference as default, but since the routine doesn't ever modify the reference it is no problem. Ideally it could be reworked better.